회원 API 수정

sujeong11 opened the pull request:
close #36 

## 작업 사항
1. `닉네임이 사용 중인지 체크하는 API` 
2. `특정 닉네임을 가지는 회원 정보를 조회하는 API`
- 회원 API swagger 문서화 추가
- API 경로 수정
  - /api/check/duplicate => /api/members/check
  - /api/users => /api/members
> 입력 값이 id가 아닌 nickname이라 RequestParam이 더 명확해보이는데 PathVariable을 사용하는 것이 더 좋을 것 같으면 말씀해주세요!

## 참고 자료

- 참고자료 (링크 등)

## 체크리스트

- [x]  PR의 제목은 팀 내 규칙을 준수하여 알맞게 작성하였는가?
- [x]  Merge 하는 브랜치가 올바른가? (`main` branch에 실수로 PR 생성 금지)
- [x]  팀의 코딩 컨벤션을 준수하는가?
- [x]  PR과 관련없는 변경사항이 들어가지는 않았는가?
- [x]  내 코드에 대한 자기 검토가 되었는가?
- [x]  Reviewers, Assignees, Lables, Project, Milestone은 적절하게 선택하였는가?
- [x]  관련한 issue를 닫아야 하는지 점검해보고 적용했는가?

sujeong11 committed:
style: #36 회원 API 경로 수정
1. 닉네임 중복체크 API, /api/check/duplicate => /api/members/check
2. 특정 닉네임을 가지는 회원정보 조회 API, /api/users => /api/members

sujeong11 committed:
feat: #36 swagger 문서화 추가

Wo-ogie reviewed (approved):
👍

Wo-ogie reviewed (changes requested):
응답 데이터인 `NicknameDuplicationCheckResponse`와 `MemberInfoGetResponse`에도 API 명세 작성할 수 있습니다!
Example 값이랑 응답 데이터 각각에 대한 설명도 같이 작성해주시면 좋을 것 같아요.
추가로, `MemberInfoGetResponse`에는 `AddressDto`도 있으니, `AddressDto`의 API 명세도 함께 작성해주시면 좋을 것 같습니다.

sujeong11 committed:
feat: #36 응답 데이터 API 명세에 추가
응답 데이터인 NicknameDuplicationCheckResponse와 MemberInfoGetResponse, MemberInfoGetResponse의 AddressDto에 대한 example 값과 설명을 API 명세에 추가

Wo-ogie commented on src/main/java/com/umc/withme/dto/member/MemberInfoGetResponse.java:
```diff
@@ -10,10 +11,19 @@
+    @Schema(example = "FEMALE", description = "회원의 성별")
     private Gender gender;
+
+    @Schema(description = "회원의 주소")
     private AddressDto address;
```
이거 아마 Swagger에서 봤을 때 설명 적용이 안될거에요.
설명을 적고 싶으면 `AddressDto`의 class level에 `@Schema(description=)`을 넣어야 하는 것 같더라구요?

sujeong11 replied:
이거 방금 확인했는데 AddressDto 눌러보면 description = "회원의 주소" 보입니다! 확인해보시고 더 문제 없으면 merge 하겠습니다!

Wo-ogie replied:
오 그럼 더 좋네요! 👍 
어차피 이건 저도 긴가민가해서 그냥 approve 했었습니당
